src/snowflake/snowpark/stored_procedure_profiler.py

```diff
@@ -122,6 +125,9 @@ def get_output(self) -> str:
         Please call this function right after the stored procedure you want to profile to avoid any error.
 
         """
+        # return empty string when profiler is not enabled to not interrupt user's code
+        if not self._is_enabled:
+            return ""
```

Comment:
nit: let's add a logging at warning level to indicate that this is not expected behavior.